cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -1328,6 +1357,10 @@ Status ConvertPySequence(PyObject* sequence_source, PyObject* mask,
 
   if (options.type == nullptr) {
     RETURN_NOT_OK(InferArrowType(seq, mask, options.from_pandas, &real_type));
+    if (options.ignore_timezone && real_type->id() == Type::TIMESTAMP) {
+      const auto& ts_type = checked_cast<TimestampType&>(*real_type);
```

Comment:
be consistent about const?

Reply:
sorry, porbably just late, but not sure what you mean here.